src/api/spec/models/workflow/step/set_flags_step_spec.rb

```diff
@@ -0,0 +1,233 @@
+        login(another_user)
+      end
+
+      it {
+        expect do
+          subject.call
+        rescue StandardError
+          Pundit::NotAuthorizedError
+        end.not_to change(Flag, :count)
+      }
+    end
+
+    context 'when user have the permissions and the project is valid' do
+      let(:scm_webhook) do
+        ScmWebhook.new(payload: {
+                         scm: 'github',
+                         event: 'pull_request',
+                         action: 'opened',
+                         pr_number: 1,
+                         target_repository_full_name: 'openSUSE/repo123',
+                         commit_sha: '123'
+                       })
+      end
+
+      context 'when one flag is given' do
+        it 'adds flag to the project' do
+          expect { subject.call }.to change(Flag, :count).by(1)
+          expect(Flag.all).to match_array([have_attributes(status: 'enable', repo: 'openSUSE_Tumbleweed', project_id: target_project.id, package_id: nil, flag: 'build')])
+        end
+      end
+
+      context 'when multiple flags are given' do
+        let(:step_instructions) do
+          {
+            flags:
+              [
+                {
+                  type: 'build',
+                  status: 'enable',
+                  project: 'home:Iggy',
+                  repository: 'openSUSE_Tumbleweed',
+                  architecture: 'x86_64'
+                },
+                {
+                  type: 'publish',
+                  status: 'enable',
+                  project: 'home:Iggy'
+                }
+              ]
+          }
+        end
+
+        it 'add flags to the project' do
+          expect { subject.call }.to change(Flag, :count).by(2)
+          expect(Flag.all).to match_array([
+                                            have_attributes(status: 'enable', repo: 'openSUSE_Tumbleweed', project_id: target_project.id, package_id: nil, flag: 'build'),
+                                            have_attributes(status: 'enable', project_id: target_project.id, package_id: nil, flag: 'publish')
+                                          ])
+        end
+      end
+    end
+
+    context 'when user have the permission and the package is valid' do
+      let!(:target_package) { create(:package, project: user.home_project, name: 'ctris-0087aa5c0549a6cc0b4c1bb324d2fa8dc665e063') }
+      let(:payload) { { scm: 'gitlab', event: 'Push Hook', commit_sha: '0087aa5c0549a6cc0b4c1bb324d2fa8dc665e063' } }
+      let(:scm_webhook) { ScmWebhook.new(payload: payload) }
+      let(:step_instructions) do
+        {
+          flags:
+            [
+              {
+                type: 'lock',
+                status: 'disable',
+                project: 'home:Iggy',
+                package: 'ctris'
+              }
+            ]
+        }
+      end
+
+      before do
+        # rubocop:disable RSpec/AnyInstance
+        allow_any_instance_of(Package).to receive(:commit_user).and_return(user)
```

Comment:
couldn't you just do it for `target_package`, i.e:

```allow(target_package).to receive(:commit_user).and_return(user)``` ?

Reply:
I know and I've tried using `allow` it doesn't solve the problem. `commit_user` always stays empty even when you add `commit_user` in `create`.

Reply:
you should login or set the `User.session = user` before creating the package. It is the same issue if you try to push something to backend in the rails console and `User.session == nil`

Reply:
Oh wow. That fixed the problem. Thanks.